### PR TITLE
[IndexTrading] ignore disabled currencies

### PR DESCRIPTION
**Pull request by GuillaumeDSM**
requires https://github.com/Drakkar-Software/OctoBot/pull/2624

### CODE REVIEW

File: Trading/Mode/index_trading_mode/tests/test_index_trading_mode.py [hunk 1]
```diff
@@ -411,7 +434,7 @@ async def test_get_rebalance_details(tools):
                     index_trading.RebalanceDetails.BUY_MORE.value: {},
                     index_trading.RebalanceDetails.REMOVE.value: {
                         "SOL": decimal.Decimal("0.3"),
-                        "ADA": decimal.Decimal("0.3")
+                        # "ADA": decimal.Decimal("0.3")  # ADA is not in traded pairs, it's not removed
```
Comment (Herklos) [inline Trading/Mode/index_trading_mode/tests/test_index_trading_mode.py:437]: 👍

File: Trading/Mode/index_trading_mode/index_trading.py [hunk 2]
```diff
@@ -448,6 +457,35 @@ def _get_filtered_traded_coins(self):
...
+            if removed_assets := [
+                asset[index_distribution.DISTRIBUTION_NAME]
+                for asset in full_distribution
+                if asset not in distribution
+            ]:
+                self.logger.info(
+                    f"Ignored {len(removed_assets)} assets {removed_assets} from configured "
+                    f"distribution as absent from traded pairs."
+                )
+            return distribution
```
Comment (Herklos) [inline Trading/Mode/index_trading_mode/index_trading.py:481]: Should we add a min assets count check like raising an error if the index that should contains 5 assets only has 2 remaining coins after the asset removal process ?

**Review by Herklos — COMMENTED** [review]
Nice feature

File: Trading/Mode/index_trading_mode/index_trading.py (hunk 2 above)
Comment (GuillaumeDSM) [reply Trading/Mode/index_trading_mode/index_trading.py:481]: currently it errors if there is less than 1 element, we could make this min configurable yep, and we set the last activity to "not enough tradable assets" or something like this

Comment (Herklos) [reply Trading/Mode/index_trading_mode/index_trading.py:481]: Yes I think we should raise something only to alert us if it happens. I'm not sure we should block the bot if the assets count is >= 1

Comment (Herklos) [reply Trading/Mode/index_trading_mode/index_trading.py:481]: It could be an automatic value of now like if more than 50% of the assets are disabled

Comment (GuillaumeDSM) [reply Trading/Mode/index_trading_mode/index_trading.py:481]: right, good idea, i'll add an error log

Comment (GuillaumeDSM) [reply Trading/Mode/index_trading_mode/index_trading.py:492]: up !

File: Trading/Mode/index_trading_mode/tests/test_index_trading_mode.py [hunk 3]
```diff
@@ -331,28 +333,56 @@ async def test_ohlcv_callback(tools):
...
         ) as get_exchange_current_time_mock:
             mode.indexed_coins = [1, 2, 3]
             await producer.ohlcv_callback("binance", "123", "BTC", "BTC/USDT", None, None)
             ensure_index_mock.assert_called_once()
+            _notify_if_missing_too_many_coins_mock.assert_called_once()
             get_exchange_current_time_mock.assert_called_once()
             assert producer._last_trigger_time == current_time * 2
 
 
+async def test_notify_if_missing_too_many_coins(tools):
```
Comment (Herklos) [inline Trading/Mode/index_trading_mode/tests/test_index_trading_mode.py:351]: 👍

**Review by Herklos — APPROVED** [review]

**Pull request merged**